control-plane-operator/controllers/hostedcontrolplane/kas/kms/aws.go

```diff
@@ -158,6 +166,9 @@ func (p *awsKMSProvider) ApplyKMSConfig(podSpec *corev1.PodSpec) error {
 	}
 	container.VolumeMounts = append(container.VolumeMounts,
 		awsKMSVolumeMounts.ContainerMounts(KasMainContainerName)...)
+
+	container.Args = append(container.Args, "--encryption-provider-config-automatic-reload=false")
```

Comment:
what does this flag do?

Reply:
It creates a file watcher in kube-apiserver that watches the encryption config file and dynamically apply changes without having to restart. It was recently introduced alongside kms v2, but if I understood the code correctly whenever the encryption type changes in Hypershift, we will rebuild the kube-apiserver anyway right? So my reasoning was to add  it as a safeguard to prevent changes to the file to be applied since they would most likely not be from the operator.

FWIW it was already added to the IBM cloud ksm code https://github.com/openshift/hypershift/blob/a11f0a1b12da74f345953f9441221e4418c314f4/control-plane-operator/controllers/hostedcontrolplane/kas/kms/ibmcloud.go#L307, but I am fine removing it here since it is kind of unrelated to the KMS v2 support.

Reply:
I see, I am fine adding it.